ng-dev/pr/takeover-pr/commit-message-filter.ts

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env node
+
+/**
+ * @license
+ * Copyright Google LLC
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+/**
+ * Script that can be passed as commit message filter to `git filter-branch --msg-filter`.
+ * The script rewrites commit messages to contain a Github instruction to close the
+ * corresponding pull request. For more details. See: https://git.io/Jv64r.
+ */
+
+main();
+
+function main() {
+  const [prNumber] = process.argv.slice(2);
+  if (!prNumber) {
+    console.error('No pull request number specified.');
+    process.exit(1);
+  }
+
+  let commitMessage = '';
+  process.stdin.setEncoding('utf8');
+  process.stdin.on('readable', () => {
+    const chunk = process.stdin.read();
+    if (chunk !== null) {
+      commitMessage += chunk;
+    }
+  });
+
+  process.stdin.on('end', () => {
+    console.info(rewriteCommitMessage(commitMessage, prNumber));
+  });
+}
+
+function rewriteCommitMessage(message: string, prNumber: string) {
+  const lines = message.split(/\n/);
+  lines.push(`Closes #${prNumber} as this was created using the \`ng-dev pr takeover\` tooling`);
```

Comment:
Is there a way to re-use the existing script? (from the autosquash strategy?)

Reply:
No, because the existing script also changes the commit message title as well.  I think its best to have it specific to the action happening.

Reply:
I see. We can avoid this script complexity and bundling with something like:

```
git commit --amend -m "$(git log --format=%B -n1)\n\nCloses #bla"

````

Reply:
This will only change the latest commit, which is not guaranteed to be included in the final commits if it is a fixup commit.

Reply:
`commit --amend` only allows you to change the most recent commit as I understand it.